pkg/cmd/kafka/create/create.go

```diff
@@ -579,48 +574,84 @@ func promptKafkaPayload(opts *options, constants *remote.DynamicServiceConstants
 
-			validMarketplaceAcctIDs := FetchValidMarketplaceAccounts(orgQuota.MarketplaceQuotas, answers.Marketplace)
+	if opts.useEnterpriseCluster {
+		/*
+			if using dedicated cluster option then get the supported sizes for this cluster
+			based on the sizes it supports the user to select the one they want to use
+			then check if the size uses <= the remaining streaming units it takes up
+			on the cluster
+		*/
+		cluster, err := getClusterDetails(opts.f, opts.selectedCluster.GetId())
+		if err != nil {
+			return nil, err
+		}
 
-			if len(validMarketplaceAcctIDs) == 1 {
-				answers.MarketplaceAcctID = validMarketplaceAcctIDs[0]
-			} else {
-				marketplaceAccountPrompt := &survey.Select{
-					Message: f.Localizer.MustLocalize("kafka.create.input.marketplaceAccountID.message"),
-					Options: validMarketplaceAcctIDs,
-				}
-				err = survey.AskOne(marketplaceAccountPrompt, &answers.MarketplaceAcctID)
-				if err != nil {
-					return nil, err
-				}
-			}
+		supportedInstanceTypes := cluster.GetSupportedInstanceTypes()
+		supportedKafkaTypes := supportedInstanceTypes.GetInstanceTypes()
+
+		// right now enterprise clusters only support one type of kafka, so we are hard
+		// checking this, this assumption may change in the future
+		if len(supportedKafkaTypes) != 1 {
+			return nil, fmt.Errorf("expected one supported instance kafka type")
 		}
-	}
 
-	marketplaceInfo := accountmgmtutil.MarketplaceInfo{
-		BillingModel:   answers.BillingModel,
-		Provider:       answers.Marketplace,
-		CloudAccountID: answers.MarketplaceAcctID,
-	}
+		sizes = supportedKafkaTypes[0].Sizes
+
+		index, err := promptUserForSizes(opts.f, &sizes)
+		if err != nil {
+			return nil, err
+		}
+
+		selectedSize := sizes[index]
+
+		if selectedSize.GetCapacityConsumed() > cluster.CapacityInformation.RemainingKafkaStreamingUnits {
+			return nil, opts.f.Localizer.MustLocalizeError("kafka.create.cluster.error.noEnoughCapacity", localize.NewEntry("Size", selectedSize.GetId()))
+		}
+
+		answers.Size = selectedSize.GetId()
+
+		if len(orgQuota.EnterpriseQuotas) == 0 {
+			return nil, fmt.Errorf("No enterprise quota available")
+		}
+		// casting to nullable string
+		var clusterIdSNS kafkamgmtclient.NullableString
+		clusterIdSNS.Set(opts.selectedCluster.ClusterId)
+
+		// need a better way to set billing model
+		var billingModelEnterprise kafkamgmtclient.NullableString
+		var enterprise = "enterprise"
+		billingModelEnterprise.Set(&enterprise)
+
+		var nullString kafkamgmtclient.NullableString
+		payload = &kafkamgmtclient.KafkaRequestPayload{
+			Name:                  answers.Name,
+			BillingModel:          billingModelEnterprise,
+			BillingCloudAccountId: nullString,
+			Marketplace:           nullString,
+			ClusterId:             clusterIdSNS,
+		}
+		// enterprise quota spec, kfm will use the default, this should be returned by the `select quota` call
+		payload.SetPlan(mapAmsTypeToBackendType(&orgQuota.EnterpriseQuotas[0]) + "." + answers.Size)
```

Comment:
Please extract the selection of `&orgQuota.EnterpriseQuotas[0]` before this line:
https://github.com/redhat-developer/app-services-cli/pull/1824/files#diff-5a5c1a033aadb8090e822a7ebdb919ec8ecbd989c46f5e84047ea197aa987293R609

i.e. closer to this meaningful comment:
https://github.com/redhat-developer/app-services-cli/pull/1824/files#diff-5a5c1a033aadb8090e822a7ebdb919ec8ecbd989c46f5e84047ea197aa987293R603-R604

So that we don't forget about updating the index when the limitation got removed.